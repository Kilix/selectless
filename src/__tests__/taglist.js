import React from 'react'
import renderer from 'react-test-renderer'

import {Select} from '../'
import TagList from '../components/tagList'
import Tag from '../components/tag'

const options = [{label: 'Paris', value: 'paris'}, {label: 'Tokyo', value: 'tokyo'}]

it('TagList', () => {
  const tree = renderer
    .create(
      <Select multi name="taglist" options={options} defaultValue={options[0]}>
        <TagList renderTag={Tag} />
      </Select>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
it('TagList custom Tag', () => {
  const tree = renderer
    .create(
      <Select multi name="taglist" options={options} defaultValue={options[0]}>
        <TagList renderTag={<Tag render={() => <div />} />} />
      </Select>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
