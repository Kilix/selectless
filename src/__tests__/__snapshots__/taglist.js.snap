// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList 1`] = `
<div>
  <input
    name="taglist[Paris]"
    type="hidden"
    value="paris"
  />
  <div
    onClick={[Function]}
  >
    <span
      onClick={[Function]}
    >
      Paris
    </span>
  </div>
</div>
`;

exports[`TagList custom Tag 1`] = `
<div>
  <input
    name="taglist[Paris]"
    type="hidden"
    value="paris"
  />
  <div
    onClick={[Function]}
  >
    <div />
  </div>
</div>
`;
